### PR TITLE
Feature/34 pipette

### LINKED ISSUE #34: Pipetten-Tool (3)
Als NutzerIn möchte ich eine beliebige Farbe aus dem Bild auswählen
können, indem ich die Stelle dieser Farbe mit der Pipette anklicke, um später damit weiterzeichnen zu können.

### PR DESCRIPTION
**Pull request by maximilian-franz**
Closes #34.

### CODE REVIEW

**Comment by kpostnov** [comment]
The implementation as it is looks good so far. However there are still some things that should be considered before this PR is to be merged:


- [x] The pipette tool icon doesn't match the actual tool yet.
- [x] `M2Pipette` should get a proper class comment just like the other `M2MenuItem` classes. 
- [x] `MorphicMonet >> chooseColor` doesn't fit into the `accessing` category.
- [x] `static` doesn't contain any methods anymore wherefore it should be removed completely.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/activateDefaultItem.st [hunk 1]
```diff
@@ -0,0 +1,7 @@
+defaults
+activateDefaultItem
+	| defaultItem |
+	defaultItem := (self submenuDict at: 'Brushes') value detect: [:item |
+			(item isM2MenuItemMorph and: [item isDefaultTool])]. "Pencil tool"
+	self menuPanel submorphs third clickItem: defaultItem.
```
Comment (kpostnov) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/activateDefaultItem.st]: On a closer look, line 6 might violate the LoD.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st [hunk 2]
```diff
@@ -1,7 +1,9 @@
 tooling
 chooseColor
-	| currentColor |
-	self world activeHand showTemporaryCursor: (CursorWithAlpha fromIcon: UiFugueIcons pipetteIcon).
+	| currentColor pipetteTool |
+	pipetteTool := self menuBrushCollection last.
+	self activeHand showTemporaryCursor:
+		(CursorWithAlpha fromIcon: pipetteTool icon).
```
Comment (kpostnov) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st]: I don't see the added value in doing this. In fact, by doing so you assume the pipette tool will always be the last element in the brush menu which might not always be the case.

**Review by kpostnov — APPROVED** [review]

File: packages/SketchMorph2-Core.package/CursorWithAlpha.extension/class/fromIcon..st [hunk 3]
```diff
@@ -0,0 +1,10 @@
+*SketchMorph2-Core
+fromIcon: aForm
+	^ (self extent: 16 asPoint depth: 32)
+		offset: 0 @ -14;
+		copyBits: (0 asPoint corner: aForm extent) 
+			from: aForm 
+			at: 0 asPoint
+			clippingBox: (0 asPoint corner: 16 asPoint) 
+			rule: Form blend
+			fillColor: nil
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/CursorWithAlpha.extension/class/fromIcon..st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2CircleBrush.class/class/icon.st [hunk 4]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons paintBrushIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2CircleBrush.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2ClearCanvas.class/class/icon.st [hunk 5]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons documentIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2ClearCanvas.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2Eraser.class/class/icon.st [hunk 6]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons eraserIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Eraser.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2ExportMorph.class/class/icon.st [hunk 7]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons blockIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2ExportMorph.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2ExportPNG.class/class/icon.st [hunk 8]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons imageExportIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2ExportPNG.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2FillArea.class/class/icon.st [hunk 9]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons paintCanIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2FillArea.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2GeometricCircle.class/class/icon.st [hunk 10]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons controlRecordIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2GeometricCircle.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2GeometricRectangle.class/class/icon.st [hunk 11]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons controlStopSquareIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2GeometricRectangle.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/defaultIconSize.st [hunk 12]
```diff
@@ -1,3 +1,3 @@
 as yet unclassified
 defaultIconSize
-	^ Preferences standardMenuFont pointSize * 2
\ No newline at end of file
+	^ 20 asPoint
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/defaultIconSize.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/defaultIconSize.st [hunk 13]
```diff
@@ -1,3 +1,3 @@
 as yet unclassified
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/defaultIconSize.st]: Uncategorized method.

File: packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/getMenuItemForTarget..st [hunk 14]
```diff
@@ -4,7 +4,7 @@ getMenuItemForTarget: anObject
 		contents: self label; 
 		target: anObject;
 		selector: self actionSelector;
-		icon: ((self perform: self iconSelector) scaledToHeight: self defaultIconSize);
+		icon: (self icon scaledToSize: self defaultIconSize);
 		isActivatable: self isActivatable;
 		isActivated: false;
 		isDefaultTool: self isDefaultTool
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/getMenuItemForTarget..st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/icon.st [hunk 15]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ self subclassResponsibility
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/icon.st [hunk 16]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/icon.st]: Uncategorized method.

File: packages/SketchMorph2-Core.package/M2ImportPNG.class/class/icon.st [hunk 17]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons imageImportIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2ImportPNG.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2Pencil.class/class/icon.st [hunk 18]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons pencilIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Pencil.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2Pipette.class/class/actionSelector.st [hunk 19]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+actionSelector
+	^ #chooseColor
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Pipette.class/class/actionSelector.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2Pipette.class/class/actionSelector.st [hunk 20]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Pipette.class/class/actionSelector.st]: Uncategorized method.

File: packages/SketchMorph2-Core.package/M2Pipette.class/class/icon.st [hunk 21]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons pipetteIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Pipette.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2Pipette.class/class/isActivatable.st [hunk 22]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+isActivatable
+	^ true
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Pipette.class/class/isActivatable.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2Pipette.class/class/label.st [hunk 23]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+label
+	^ 'Pipette'
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Pipette.class/class/label.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2Pipette.class/class/icon.st [hunk 24]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Pipette.class/class/icon.st]: Uncategorized method.

File: packages/SketchMorph2-Core.package/M2Pipette.class/class/isActivatable.st [hunk 25]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Pipette.class/class/isActivatable.st]: Uncategorized method.

File: packages/SketchMorph2-Core.package/M2Pipette.class/class/label.st [hunk 26]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Pipette.class/class/label.st]: Uncategorized method.

File: packages/SketchMorph2-Core.package/M2RectangleBrush.class/class/icon.st [hunk 27]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons paintBrushIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2RectangleBrush.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2RectangleBrush.class/class/icon.st [hunk 28]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2RectangleBrush.class/class/icon.st]: Uncategorized method.

File: packages/SketchMorph2-Core.package/M2TriangleBrush.class/class/icon.st [hunk 29]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons paintBrushIcon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2TriangleBrush.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2Undo.class/class/icon.st [hunk 30]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+icon
+	^ MorphicMonetIcons arrowCurve180Icon
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Undo.class/class/icon.st]: Missing final newline.

File: packages/SketchMorph2-Core.package/M2TriangleBrush.class/class/icon.st [hunk 31]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2TriangleBrush.class/class/icon.st]: Uncategorized method.

File: packages/SketchMorph2-Core.package/M2Undo.class/class/icon.st [hunk 32]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Undo.class/class/icon.st]: Uncategorized method.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuBrushCollection..st [hunk 33]
```diff
@@ -0,0 +1,3 @@
+accessing
+menuBrushCollection: anObject
+	menuBrushCollection := anObject
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuBrushCollection..st]: Missing final newline.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuGeometricCollection..st [hunk 34]
```diff
@@ -0,0 +1,3 @@
+accessing
+menuGeometricCollection: anObject
+	menuGeometricCollection := anObject
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuGeometricCollection..st]: Missing final newline.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuMiscCollection..st [hunk 35]
```diff
@@ -0,0 +1,3 @@
+accessing
+menuMiscCollection: anObject
+	menuMiscCollection := anObject
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuMiscCollection..st]: Missing final newline.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/submenuDict..st [hunk 36]
```diff
@@ -0,0 +1,3 @@
+accessing
+submenuDict: anObject
+	submenuDict := anObject
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/submenuDict..st]: Missing final newline.

File: packages/SketchMorph2-Core.package/MorphicMonetIcons.class/class/license.st [hunk 37]
```diff
@@ -0,0 +1,7 @@
+license
+license
+	^ 'Fugue Icons 3.0 (shadowless)
+		© 2008-2010 Yusuke Kamiyamane
+		Website: http://p.yusukekamiyamane.com
+		License: http://creativecommons.org/licenses/by/3.0'
+
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/MorphicMonetIcons.class/class/license.st]: Final newline is not empty.

File: packages/SketchMorph2-Core.package/M2CircleBrush.class/class/icon.st [hunk 38]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2CircleBrush.class/class/icon.st]: Method is unclassified.

File: packages/SketchMorph2-Core.package/M2ClearCanvas.class/class/icon.st [hunk 39]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2ClearCanvas.class/class/icon.st]: Method is unclassified.

File: packages/SketchMorph2-Core.package/M2Eraser.class/class/icon.st [hunk 40]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2Eraser.class/class/icon.st]: Method is unclassified.

File: packages/SketchMorph2-Core.package/M2ExportMorph.class/class/icon.st [hunk 41]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2ExportMorph.class/class/icon.st]: Method is unclassified.

File: packages/SketchMorph2-Core.package/M2ExportPNG.class/class/icon.st [hunk 42]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2ExportPNG.class/class/icon.st]: Method is unclassified.

File: packages/SketchMorph2-Core.package/M2FillArea.class/class/icon.st [hunk 43]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2FillArea.class/class/icon.st]: Method is unclassified.

File: packages/SketchMorph2-Core.package/M2GeometricCircle.class/class/icon.st [hunk 44]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2GeometricCircle.class/class/icon.st]: Method is unclassified.

File: packages/SketchMorph2-Core.package/M2GeometricRectangle.class/class/icon.st [hunk 45]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2GeometricRectangle.class/class/icon.st]: Method is unclassified.

File: packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/defaultIconSize.st (hunk 12 above)
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/defaultIconSize.st]: Why this change? The old version seems to have a point, having the icon size being dependent on the font size of Squeak.

File: packages/SketchMorph2-Core.package/M2ImportPNG.class/class/icon.st [hunk 46]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2ImportPNG.class/class/icon.st]: Method is unclassified.

File: packages/SketchMorph2-Core.package/M2Pencil.class/class/icon.st [hunk 47]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2Pencil.class/class/icon.st]: Method is unclassified.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/activateDefaultItem.st (hunk 1 above)
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/activateDefaultItem.st]: I don't think it is a good idea to use the third panel and expecting the paint tool selection panel there.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuBrushCollection..st [hunk 48]
```diff
@@ -0,0 +1,3 @@
+accessing
+menuBrushCollection: anObject
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuBrushCollection..st]:
```suggestion
menuBrushCollection: aCollection
```

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuGeometricCollection..st [hunk 49]
```diff
@@ -0,0 +1,3 @@
+accessing
+menuGeometricCollection: anObject
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuGeometricCollection..st]:
```suggestion
menuGeometricCollection: aCollection
```

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st [hunk 50]
```diff
@@ -0,0 +1,9 @@
+tooling
+chooseColor
+	| currentColor |
+	self world activeHand showTemporaryCursor: 
+		(CursorWithAlpha fromIcon: UiFugueIcons pipetteIcon).
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st]: I don't think we should use `UiFugueIcons` since we cannot ensure the package will be installed.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st [hunk 51]
```diff
@@ -0,0 +1,9 @@
...
+	[Sensor redButtonPressed] whileFalse:
+		[currentColor := (Display colorAt: Sensor cursorPoint)].
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st]: Why are we fetching the color all the time? Wouldn't it be sufficient to fetch it one time after the loop?

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st [hunk 52]
```diff
@@ -0,0 +1,9 @@
+tooling
+chooseColor
+	| currentColor |
+	self world activeHand showTemporaryCursor: 
+		(CursorWithAlpha fromIcon: UiFugueIcons pipetteIcon).
+	[Sensor redButtonPressed] whileFalse:
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st]: I do not like this solution at all. It
- doesn't allow the pipette to be tested.
- uses up one CPU thread. These things don't grow on trees.
- stops the execution of everything else in Squeak. I do believe that there could be 100 things that break that we just don't notice.
- creates artifacts when resizing Squeak while the pipette is active: 
image

Note that this list may not be complete.

What about snapshotting the world and displaying it until a mouse down/up event gets triggered?

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuGeometricCollection.st [hunk 53]
```diff
@@ -1,9 +1,8 @@
...
 menuGeometricCollection
-	^ (OrderedCollection new)
-			add: (M2FillArea getMenuItemForTarget: self backend);
-			add: (M2GeometricCircle getMenuItemForTarget: self backend);
-			add: (M2GeometricRectangle getMenuItemForTarget: self backend);
-			add: (M2FillGeometricForm getMenuItemForTarget: self backend);
-			yourself
-		
\ No newline at end of file
+	^ menuGeometricCollection ifNil: [(OrderedCollection new)
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuGeometricCollection.st]:
```suggestion
	^ menuGeometricCollection ifNil: [menuGeometricCollection := (OrderedCollection new)
```

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuMiscCollection..st [hunk 54]
```diff
@@ -0,0 +1,3 @@
+accessing
+menuMiscCollection: anObject
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuMiscCollection..st]:
```suggestion
menuMiscCollection: aCollection
```

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuMiscCollection.st [hunk 55]
```diff
@@ -1,9 +1,9 @@
-static
+accessing
 menuMiscCollection
-	^ (OrderedCollection new)
+	^ menuMiscCollection ifNil: [(OrderedCollection new)
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/menuMiscCollection.st]:
```suggestion
	^ menuMiscCollection ifNil: [menuMiscCollection := (OrderedCollection new)
```

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/submenuDict..st [hunk 56]
```diff
@@ -0,0 +1,3 @@
+accessing
+submenuDict: anObject
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/submenuDict..st]:
```suggestion
submenuDict: aDictionary
```

File: packages/SketchMorph2-Core.package/MorphicMonet.class/properties.json [hunk 57]
```diff
@@ -12,7 +12,11 @@
 		"topPanel",
 		"colorPanel",
 		"colorPalette",
-		"colorPicker" ],
+		"colorPicker",
+		"menuBrushCollection",
+		"menuGeometricCollection",
+		"menuMiscCollection",
+		"submenuDict" ],
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/properties.json]: Could you order these alphabetically?

**Review by ClFeSc — CHANGES_REQUESTED** [review]
I didn't check that the icons are copied correctly. Everything I saw in Squeak looks fine so I assume the errors, if any, are not dramatic.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/activateDefaultItem.st (hunk 1 above)
Comment (maximilian-franz) [reply packages/SketchMorph2-Core.package/MorphicMonet.class/instance/activateDefaultItem.st]: I'm not sure if there is another/better way to access it due to the way it is created. (`MorphicMonet>>loadMenu`)

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st (hunk 52 above)
Comment (maximilian-franz) [reply packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st]: If we follow the snapshot approach, I'd prefer the invisible morph technique since it does basically the same and doesn't "freeze" the frame like a snapshot would.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/loadMenu.st [hunk 58]
```diff
@@ -3,9 +3,8 @@ loadMenu
...
-		buttonCollection do: [:item |
-			submenu addMenuItem: item.
-			(item isM2MenuItemMorph and: [item isDefaultTool]) ifTrue: [submenu clickItem: item]].
+		buttonCollection do: [:item | submenu addMenuItem: item].
 		self applyDefaultSettingsTo: submenu.
 		submenu beSticky.
-		self menuPanel addMorph: submenu]
\ No newline at end of file
+		self menuPanel addMorph: submenu].
+	    self activateDefaultItem
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/loadMenu.st]: This line should be indented one tab less.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/activateDefaultItem.st (hunk 1 above)
Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/MorphicMonet.class/instance/activateDefaultItem.st]: Hm. Maybe we could check which submorph contains `defaultItem` and then select it?

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st (hunk 52 above)
Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st]: I'd be fine with this too. But I think, the other one would be more like other software. When you make a screenshot with e.g. SnippingTool, it also freezes the background. But as long as we don't have this loop, I'm happy.

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st [hunk 59]
```diff
@@ -1,9 +1,8 @@
...
-	| currentColor |
+	| probePosition |
 	self world activeHand showTemporaryCursor: 
-		(CursorWithAlpha fromIcon: UiFugueIcons pipetteIcon).
-	[Sensor redButtonPressed] whileFalse:
-		[currentColor := (Display colorAt: Sensor cursorPoint)].
-	self setColor: currentColor.
+		(CursorWithAlpha fromIcon: MorphicMonetIcons pipetteIcon).
+	[Sensor redButtonPressed] whileFalse: [probePosition := Sensor cursorPoint].
+	self setColor: (Display colorAt: probePosition).
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st]:
```suggestion
	[Sensor redButtonPressed] whileFalse: [].
	self setColor: (Display colorAt: Sensor cursorPoint).
```

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/defaultCanvasPanelBorder.st [hunk 60]
```diff
@@ -1,3 +1,3 @@
 defaults
 defaultCanvasPanelBorder
-	^ 4 @ 4
\ No newline at end of file
+	^ 4 @ 4
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/MorphicMonet.class/instance/defaultCanvasPanelBorder.st]:
```suggestion
	^ 4 asPoint
```

**Review by ClFeSc — CHANGES_REQUESTED** [review]
~~Some newlines still missing.~~ Newlines look good after commit [ccc714d](https://github.com/hpi-swa-teaching/SketchMorph2/pull/47/commits/ccc714d7b6fc4f0ae38e98755273a3865ca2d4b0).

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/activateDefaultItem.st (hunk 1 above)
Comment (maximilian-franz) [reply packages/SketchMorph2-Core.package/MorphicMonet.class/instance/activateDefaultItem.st]: Yeah that could work.

Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/MorphicMonet.class/instance/activateDefaultItem.st]: Due to the short time we have remaining we can leave this as it is, in my opinion.

**Review by ClFeSc — COMMENTED** [review]
Apart from the looping in the color selection I'm happy with the changes.

**Review by kpostnov — APPROVED** [review]

**Review by johannes-poetzsch — APPROVED** [review]

File: packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st (hunk 52 above)
Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/MorphicMonet.class/instance/chooseColor.st]: We'll leave this as a separate issue.

**Review by ClFeSc — APPROVED** [review]
I agree to leave the loop for now.

**Review by jannis-hpi — APPROVED** [review]

**Review by DevSchmidtchen — APPROVED** [review]
